Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm

```diff
@@ -499,6 +504,9 @@ static OSStatus sampleCallback(CMBufferRef buffer, void* refcon)
     if (m_compressedSampleQueue.isEmpty())
         return;
 
+    if (m_hasOutOfOrderFrames && m_compressedSampleQueue.size() < MaximumSlidingWindowLength && m_compressedSampleQueueSize >= MaximumSlidingWindowLength)
```

Comment:
This is not easy to read, maybe we should rename `m_compressedSampleQueueSize` to `m_compressedSamplesCount` maybe.